protocol/xmrmaker/backend_offers.go

```diff
@@ -40,16 +40,9 @@ func (b *Instance) GetOffers() []*types.Offer {
 }
 
 // ClearOffers clears all offers.
-// If the offer list is empty, it clears all offers.
 func (b *Instance) ClearOffers(offerIDs []types.Hash) error {
-	l := len(offerIDs)
-	if l == 0 {
-		err := b.offerManager.ClearAllOffers()
-		if err != nil {
-			return err
-		}
```

Comment:
I don't think we intended to run `ClearOfferIDs` after already running `ClearAllOffers`.